readthedocs/rtd_tests/tests/test_build_forms.py

```diff
@@ -147,3 +157,182 @@ def test_resources_are_deleted_when_version_is_inactive(
+                    "privacy_level": PUBLIC,
+                },
+                instance=version,
+                project=self.project,
+            )
+            assert not form.is_valid()
+            assert "slug" in form.errors
+
+    def test_change_slug_already_in_use(self):
+        version_one = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="one",
+        )
+        version_two = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="two",
+        )
+
+        form = VersionForm(
+            {
+                "slug": version_two.slug,
+                "active": True,
+                "privacy_level": PUBLIC,
+            },
+            instance=version_one,
+            project=self.project,
+        )
+        assert not form.is_valid()
+        assert "slug" in form.errors
+
+    def test_cant_change_slug_machine_created_versions(self):
+        version = self.project.versions.get(slug="latest")
+        assert version.machine
+
+        form = VersionForm(
+            {
+                "slug": "change",
+                "active": True,
+                "privacy_level": PUBLIC,
+            },
+            instance=version,
+            project=self.project,
+        )
+        assert form.is_valid()
+        version.refresh_from_db()
+        assert version.slug == "latest"
+
+    @mock.patch("readthedocs.builds.models.trigger_build")
+    @mock.patch("readthedocs.projects.tasks.search.remove_search_indexes")
+    @mock.patch("readthedocs.projects.tasks.utils.remove_build_storage_paths")
+    def test_clean_resources_when_changing_slug_of_active_version(
+        self, remove_build_storage_paths, remove_search_indexes, trigger_build
+    ):
+        version = get(
+            Version,
+            project=self.project,
+            active=True,
+            slug="slug",
+        )
+
+        self.client.force_login(self.user)
+        url = reverse(
+            "project_version_detail", args=(version.project.slug, version.slug)
+        )
+        r = self.client.post(
+            url,
+            data={
+                "slug": "change-me",
+                "active": True,
+                "privacy_level": PUBLIC,
+            },
+        )
+        assert r.status_code == 302
+
+        version.refresh_from_db()
+        assert version.slug == "change-me"
+
+        remove_build_storage_paths.delay.assert_called_once_with(
+            [
+                "html/project/slug",
+                "pdf/project/slug",
+                "epub/project/slug",
+                "htmlzip/project/slug",
+                "json/project/slug",
+                "diff/project/slug",
+            ]
+        )
+        remove_search_indexes.delay.assert_called_once_with(
+            project_slug=version.project.slug,
+            version_slug="slug",
+        )
+        trigger_build.assert_called_once()
```

Comment:
Can't we use `assert_called_once_with` here as well to check the right project and version was triggered?